Mathlib/RingTheory/Valuation/RankOne.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Data.Real.NNReal
+import Mathlib.RingTheory.Valuation.Basic
+
+/-!
+# Rank one valuations
+
+We define rank one valuations.
+
+## Main Definitions
+* `IsRankOne` : A valuation has rank one if it is nontrivial and its image is contained in `ℝ≥0`.
+
+## Tags
+
+valuation, rank one
+-/
+
+noncomputable section
+
+open Function Multiplicative
+
+open scoped NNReal
+
+variable {R : Type*} [Ring R] {Γ₀ : Type*} [LinearOrderedCommGroupWithZero Γ₀]
+
+namespace Valuation
+
+/-- A valuation has rank one if it is nontrivial and its image is contained in `ℝ≥0`. -/
+class IsRankOne (v : Valuation R Γ₀) where
```

Comment:
In this way `IsRankOne` is not a `Prop` right? I am not sure, but maybe we can redefine it asking the existence of `hom` (and then using choice to get one if we need).

Reply:
Defining it as a `Prop` was my first approach, but I decided it against it because then I lose control of normalizations. In [this file](https://github.com/mariainesdff/norm_extensions_lean_4/blob/master/NormExtensions/NormedValued.lean), I define a dictionary between rank one valuations and nontrivial nonarchimedean norms, and I want to make sure that the normalized p-adic valuation corresponds to the normalized p-adic norm under this dictionary, but this isn't true if I can't control the inclusion of `WithZero (Multiplicative Z)` in `NNReal`. (I guess this may be possible renormalizing the map given by the existential, but then the dictionary will get more complicated).

Reply:
OK, if you have reasons to do it this way no problems. I am not sure about the name, even if we use `IsBlah` for non `Prop` in other places (but people regularly complain on Zulip).

Reply:
@kbuzzard what do you think? (Asking because you were involved in the Zulip discussion)

Reply:
Yes I think `IsRankOne` is a recipe for confusion because people will assume it's a Prop. I don't think there are so many other places where we're using `Is` for a non-Prop: `IsROrC` got changed to `RCLike` for example.  Very happy with `RankOne`.